cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs

```diff
@@ -1700,6 +1701,31 @@ runQueryDRepStakeDistribution
     writeOutput mOutFile $
       Map.assocs drepStakeDistribution
 
+runQuerySPOStakeDistribution
+  :: Cmd.QuerySPOStakeDistributionCmdArgs era
+  -> ExceptT QueryCmdError IO ()
+runQuerySPOStakeDistribution
+  Cmd.QuerySPOStakeDistributionCmdArgs
+    { Cmd.eon
+    , Cmd.nodeSocketPath
+    , Cmd.consensusModeParams
+    , Cmd.networkId
+    , Cmd.spoHashSources = spoHashSources'
+    , Cmd.target
+    , Cmd.mOutFile
+    } = conwayEraOnwardsConstraints eon $ do
+    let localNodeConnInfo = LocalNodeConnectInfo consensusModeParams networkId nodeSocketPath
+        spoFromSource = firstExceptT QueryCmdSPOKeyError . readSPOCredential
+        spoHashSources = case spoHashSources' of
+          All -> []
+          Only l -> l
+
+    spos <- Set.fromList <$> mapM spoFromSource spoHashSources
```

Comment:
```suggestion
    spos <- fromList <$> mapM spoFromSource spoHashSources
```

Reply:
Hm, it seems that is necessary there. Also we are importing `Map` too, so it could be confusing if we just imported it unqualified. We could use the `toList` from `Foldable`, but for `fromList` I don't know a generic one